base_layer/wallet/src/output_manager_service/handle.rs

```diff
@@ -101,10 +105,16 @@ pub enum OutputManagerRequest {
     CancelTransaction(TxId),
     GetSpentOutputs,
     GetUnspentOutputs,
+    GetOutputsBy(OutputBackendQuery),
     GetInvalidOutputs,
     ValidateUtxos,
     RevalidateTxos,
     CreateCoinSplit((MicroTari, usize, MicroTari, Option<u64>)),
+    CreateCoinJoin {
+        target_amount: MicroTari,
+        fee_per_gram: MicroTari,
+        commitments: Option<Vec<PublicKey>>,
```

Comment:
Rather use an empty vec to represent "None"
```suggestion
        commitments: Vec<PublicKey>,
```